catalyst/dl/utils/swa.py

```diff
@@ -0,0 +1,62 @@
+from typing import List
+from collections import OrderedDict
+import glob
+import os
+from pathlib import Path
+
+import torch
+
+
+def average_weights(state_dicts: List[dict]) -> OrderedDict:
+    """
+    Averaging of input weights.
+
+    Args:
+        state_dicts (List[dict]): Weights to average
+
+    Returns:
+        Averaged weights
+    """
+    # source https://gist.github.com/qubvel/70c3d5e4cddcde731408f478e12ef87b
+
+    average_dict = OrderedDict()
+    for k in state_dicts[0].keys():
+        average_dict[k] = torch.div(
+            sum(state_dict[k] for state_dict in state_dicts), len(state_dicts),
+        )
+    return average_dict
```

Comment:
Hi, what do you think about adding state dict key check mechanics (raise an error when key is missing)?
Probably it will be informative to know that something wrong with `<checkpoint-file-name>.pth`.

For example you could check this - https://gist.github.com/Ditwoo/5de19670d9946c80916dee75e93ef545

Reply:
@Ditwoo your implementation also support fp16 variant, is it correct?

Reply:
@ivashnyov could you please check it out?